homeassistant/components/mqtt/__init__.py

```diff
@@ -364,8 +364,15 @@ async def async_setup_reload_service() -> None:
 
         async def _reload_config(call: ServiceCall) -> None:
             """Reload the platforms."""
-            # Fetch updated manual configured items and validate
-            config_yaml = await async_integration_yaml_config(hass, DOMAIN) or {}
+            # Fetch updated manually configured items and validate
+            if (
+                config_yaml := await async_integration_yaml_config(hass, DOMAIN)
```

Comment:
We should maybe make `async_integration_yaml_config` raise instead (it will require an option to `async_integration_yaml_config` to raise), then we can trap errors here and print a single error message.
If there's an error now, there will first be an error log from `async_process_component_config`, then we log here again.

Reply:
`async_process_component_config` has lots of cases for which it logs an error. If we want this helper to throw we should open a separate PR for that, so we can use it later here.